Codecademy/client-modules
# feat(Styleguide): Docs Navigation - GM-133

## Overview

This adds a bunch of navigation features to our storybook, including:
- Table of Contents on every index page with enriched meta information (components and status).
- Breadcrumbs at every page level, to make it easier to figure out what subjection you are in at a glance.
- Adds a `subtitle` parameter to each story which is shown at the top of the document and in the sections card in the table of contents.

Stories:
- Each component has a filled out `subtitle` string as a short description.  This can be any valid markdown and formatted correctly.
- Edited some of the story hierarchy to improve structure.

### Usage

Adding a table of contents is as simple as dropping docs block in any page:
```tsx
import { TableOfContents } from '~styleguide/blocks';

<TableOfContents />
```

The component will automatically find all relevant stories/sections to the current context and display their subsections with some added content.

### Screenshots

**Breadcrumbs**
![localhost_6006__path=_docs_gamut--page (1)](https://user-images.githubusercontent.com/52927224/104066447-70f33800-51cf-11eb-9288-4a516a0b92fb.png)

**Top Level ToC**
![localhost_6006__path=_docs_gamut--page](https://user-images.githubusercontent.com/52927224/104066448-718bce80-51cf-11eb-8815-a9f9c1eff7c2.png)


* Adds `TableOfContents` Doc Block to our styleguide storybook components.
* Adds breadcrumb to each Docs page.
* Adds a subtitle parameter to Story parameters so descriptions are available.


### PR Checklist

- [x] Related to JIRA ticket: GM-133

### Testing

1. Visit https://5ff8d1826b57e70fd9e52bcc--gamut-preview.netlify.app
2. Click through some of the content sections.
3. And navigate back up using the `Breadcrumbs` at the top of the page.
4. Ensure things make sense.
